probdist/normal.go

```diff
@@ -1,43 +1,142 @@
 package stats
 
 import (
+	"errors"
 	"math"
 )
 
-// Normal is used to represent the normal distribution parameters
-// Mu is the mean
-// Sigma is the standard deviation
+// Normal is used to represent the normal distribution parameters.
+// Mu is the mean.
+// Sigma is the standard deviation.
 type Normal struct {
 	Mu    float64
 	Sigma float64
 }
 
-// NewNormal is used to initialize normal parameters
-func NewNormal() Normal {
-	return Normal{0.0, 1.0}
+// NewNormal is used to initialize normal parameters. What is different from
+// Normal type is that here the parameters are validated.
+// Mu must be a real number (+ or -) and Sigma > 0.
+func NewNormal(mu float64, sigma float64) (Normal, error) {
+	if sigma <= 0 {
+		return Normal{0, 1.0}, errors.New("stats: invalid Normal parameters. Check Sigma > 0")
+	}
+	return Normal{Mu: mu, Sigma: sigma}, nil
 }
 
-// CDF returns the cumulative distribution function output of the normal distribution for a given x
+// CDF returns the cumulative distribution function output of the normal distribution for a given x.
 func (norm Normal) CDF(x float64) float64 {
 	return 0.5 * (1 + math.Erf((x-norm.Mu)/(math.Sqrt(2)*norm.Sigma)))
 }
 
-// Mean returns the mean of the normal distribution
+// Mean returns the mean of the normal distribution.
 func (norm Normal) Mean() float64 {
 	return norm.Mu
 }
 
-// Median returns the median of the normal distribution
+// Median returns the median of the normal distribution.
 func (norm Normal) Median() float64 {
 	return norm.Mu
 }
 
-// StdDev returns the standard deviation of the normal distribution
+// StdDev returns the standard deviation of the normal distribution.
 func (norm Normal) StdDev() float64 {
 	return norm.Sigma
 }
 
-// Variance returns the Variance of the normal distribution
+// Variance returns the Variance of the normal distribution.
 func (norm Normal) Variance() float64 {
 	return math.Pow(norm.Sigma, 2.0)
 }
+
+//Data from: https://www.itl.nist.gov/div898/handbook/eda/section3/eda3671.htm.
+var zStandardNormal statsTable = statsTable{
+	[][]float64{
+		[]float64{0.00000, 0.00399, 0.00798, 0.01197, 0.01595, 0.01994, 0.02392, 0.02790, 0.03188, 0.03586},
```

Comment:
I think that it is not necessary to repeat `[]float64` in every member of the array.

If this table has a fixed number of rows and columns, better use an array of array than a slice of slices, so the compiler can optimize it and even store it in the stack.